core/shared/src/main/scala/monocle/function/Snoc.scala

```diff
@@ -57,12 +57,12 @@ object Snoc extends SnocFunctions {
       case (init, last) => init :+ last
     }
   )
-
-  implicit def streamSnoc[A]: Snoc[Stream[A], A] = Snoc(
-    Prism[Stream[A], (Stream[A], A)]( s =>
+// replace with LazyList
+  implicit def lazyListSnoc[A]: Snoc[LazyList[A], A] = Snoc(
```

Comment:
I am not sure we can change this to LazyList yet because 2.12 doesn't have LazyList (if I am not mistaken)